### PR TITLE
add AnnotationDAOTest

### PR DESCRIPTION
**Pull request by justinvdk**
This PR adds simple tests for the AnnotationDAO class

### CODE REVIEW

File: PL2/PL2-shared/src/test/java/nl/tudelft/pl2016gr2/dao/AnnotationDAOTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    testAnnotations.add(a1);
+    testAnnotations.add(a2);
+    testAnnotations.add(a3);
+    testAnnotations.add(a4);
+
+    dao = new AnnotationDAO(testAnnotations);
+  }
+
+  @Test
+  public void testgetAnnotation() throws Exception {
```
Comment (wouterjsmit) [inline PL2/PL2-shared/src/test/java/nl/tudelft/pl2016gr2/dao/AnnotationDAOTest.java]: Please use camelcase for the after '`test`'. Please do so for every method.

File: PL2/PL2-shared/src/test/java/nl/tudelft/pl2016gr2/dao/AnnotationDAOTest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    Collection<Annotation> annotations = dao.getAllAnnotations();
+
+    assertEquals(4, annotations.size());
+    Stream.of(a1, a2, a3, a4).forEach(annotation -> {
+      assertTrue(annotations.contains(annotation));
+    });
+
+  }
+
```
Comment (wouterjsmit) [inline PL2/PL2-shared/src/test/java/nl/tudelft/pl2016gr2/dao/AnnotationDAOTest.java:61]: Remove the whitespace between successive closing brackets.

File: PL2/PL2-shared/src/test/java/nl/tudelft/pl2016gr2/dao/AnnotationDAOTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    assertSame(dao.getAnnotation("a1"), a1);
+    assertSame(dao.getAnnotation("a2"), a2);
+    assertSame(dao.getAnnotation("a3"), a3);
+    assertSame(dao.getAnnotation("a4"), a4);
+
+  }
+
+  @Test
+  public void testgetAllAnnotations() throws Exception {
+
```
Comment (wouterjsmit) [inline PL2/PL2-shared/src/test/java/nl/tudelft/pl2016gr2/dao/AnnotationDAOTest.java]: Unnecessary whitespace.

File: PL2/PL2-shared/src/test/java/nl/tudelft/pl2016gr2/dao/AnnotationDAOTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    testAnnotations.add(a2);
+    testAnnotations.add(a3);
+    testAnnotations.add(a4);
+
+    dao = new AnnotationDAO(testAnnotations);
+  }
+
+  @Test
+  public void testgetAnnotation() throws Exception {
+
```
Comment (wouterjsmit) [inline PL2/PL2-shared/src/test/java/nl/tudelft/pl2016gr2/dao/AnnotationDAOTest.java]: Unnecessary whitespace.

File: PL2/PL2-shared/src/test/java/nl/tudelft/pl2016gr2/dao/AnnotationDAOTest.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  Annotation a2 = new Annotation();
+  Annotation a3 = new Annotation();
+  Annotation a4 = new Annotation();
+
+  /**
+   * Setups up the {@link #dao} with the above test annotations.
+   */
+  @Before
+  public void setup() {
+
```
Comment (wouterjsmit) [inline PL2/PL2-shared/src/test/java/nl/tudelft/pl2016gr2/dao/AnnotationDAOTest.java]: Unnecessary whitespace.

File: PL2/PL2-shared/src/test/java/nl/tudelft/pl2016gr2/dao/AnnotationDAOTest.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+public class AnnotationDAOTest {
+
+  AnnotationDAO dao;
+  Annotation a1 = new Annotation();
+  Annotation a2 = new Annotation();
+  Annotation a3 = new Annotation();
+  Annotation a4 = new Annotation();
+
+  /**
+   * Setups up the {@link #dao} with the above test annotations.
```
Comment (wouterjsmit) [inline PL2/PL2-shared/src/test/java/nl/tudelft/pl2016gr2/dao/AnnotationDAOTest.java]: `Setups up -> Sets up`

Also, a hyperlink to an object that has no JavaDoc is pointless. use `<code></code>` instead, or add javadoc.

File: PL2/PL2-shared/src/test/java/nl/tudelft/pl2016gr2/dao/AnnotationDAOTest.java (hunk 1 above)
Comment (justinvdk) [inline PL2/PL2-shared/src/test/java/nl/tudelft/pl2016gr2/dao/AnnotationDAOTest.java]: This was a typo, thanks
